### PR TITLE
STM32F401 temperature improvements

### PR DESCRIPTION
**Pull request by mbolivar**

### CODE REVIEW

**Comment by mbolivar** [comment]
Carbon and Nitrogen are now changing by roughly the same amount of degrees centigrade when I point the hair dryer at them now.

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
Sanity Check Succeeded.

File: drivers/sensor/stm32/temp_stm32f401x.c [hunk 1]
```diff
@@ -195,10 +195,9 @@ static int temp_stm32f401x_init(struct device *dev)
 	 *   mutually exclusive, and VBAT has precedence).
 	 */
 	tmp = adc_common->CCR;
-	tmp |= (ADC_CCR_TSVREFE << ADC_CCR_TSVREFE_Pos);
-	tmp &= ~(ADC_CCR_ADCPRE << ADC_CCR_ADCPRE_Pos);
-	tmp |= (ADC_PRESCALER_PCLK_DIV_8 << ADC_CCR_ADCPRE_Pos);
-	tmp &= ~(ADC_CCR_VBATE << ADC_CCR_VBATE_Pos);
+	tmp &= ~(ADC_CCR_VBATE | ADC_CCR_ADCPRE);
+	tmp |= ((ADC_PRESCALER_PCLK_DIV_8 << ADC_CCR_ADCPRE_Pos) |
```
Comment (mike-scott) [inline drivers/sensor/stm32/temp_stm32f401x.c:222]: Is ADC_PRESCALER_PCLK_DIV_8 not shifted by ADC_CCR_ADCPRE_Pos? Like the other macros?

File: drivers/sensor/stm32/temp_stm32f401x.c [hunk 2]
```diff
@@ -135,10 +144,9 @@ static int temp_stm32f401x_sample_fetch(struct device *dev,
 	return 0;
 }
 
-static inline float stm32f401x_temp_c(uint32_t v_sense)
+static inline float stm32f401x_temp_c(float v_sense)
 {
-	float sense = (float)v_sense;
-	return ((sense - STM32F401_V25) / STM32F401_AVG_SLOPE + 25.0f);
+	return ((v_sense - STM32F401_V25) / STM32F401_AVG_SLOPE + 25.0f);
```
Comment (mike-scott) [inline drivers/sensor/stm32/temp_stm32f401x.c]: Can we add a comment on what the value of 25.0f is doing?

File: drivers/sensor/stm32/temp_stm32f401x.c (hunk 1 above)
Comment (mbolivar) [reply drivers/sensor/stm32/temp_stm32f401x.c:222]: Yeah, that one is defined up earlier in the file.

The way it works is that stm32f401xe.h provides:

```
#define ADC_REG_FIELD_Msk 0xvalue
#define ADC_REG_FIELD_Pos bit_shift_value
#define ADC_REG_FIELD (ADC_REG_FIELD_Msk << ADC_REG_FIELD_Pos)
```
So for encoded values that I define myself in this file, I have to shift them.

File: drivers/sensor/stm32/temp_stm32f401x.c (hunk 2 above)
Comment (mbolivar) [reply drivers/sensor/stm32/temp_stm32f401x.c]: Sure -- I'm just copy/pasting this from the reference manual. I'll add a pointer to the relevant section.

File: drivers/sensor/stm32/temp_stm32f401x.c (hunk 1 above)
Comment (mike-scott) [reply drivers/sensor/stm32/temp_stm32f401x.c:222]: Make sense, thanks

File: drivers/sensor/stm32/temp_stm32f401x.c (hunk 2 above)
Comment (mike-scott) [reply drivers/sensor/stm32/temp_stm32f401x.c]: I was just curious and won't block the merge over this.  If you can add it easily that's great.

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Review by mike-scott — APPROVED** [review]
LGTM

**Comment by gatek33p3r** [comment]
Sanity Check Succeeded.

**Pull request merged**